tracer/src/Datadog.Trace/Telemetry/TelemetrySettings.cs

```diff
@@ -143,8 +143,14 @@ public static TelemetrySettings FromSource(IConfigurationSource source, IConfigu
                                    .Value;
 
             var dependencyCollectionEnabled = config.WithKeys(ConfigurationKeys.Telemetry.DependencyCollectionEnabled).AsBool(true);
-            // Currently disabled, will be flipped to true in later versions as part of the rollout
-            var v2Enabled = config.WithKeys(ConfigurationKeys.Telemetry.V2Enabled).AsBool(false);
+
+            // TODO: Remove when we rollout telemetry v2. As of now, it will be activated by default only in AAS.
+            // we already fetch this, so this will overwrite the telemetry.... no biggy as temporary
+            var isRunningInAzureAppService = config
+                                        .WithKeys(ConfigurationKeys.AzureAppService.AzureAppServicesContextKey)
+                                        .AsBool(false);
+
+            var v2Enabled = config.WithKeys(ConfigurationKeys.Telemetry.V2Enabled).AsBool(defaultValue: isRunningInAzureAppService);
```

Comment:
Interesting, I was thinking we would just set the variable in the AAS extension, that way we could easily push out another AAS extension release disabling them if necessary. It would also mean we could push out the change independently of the new release, which further would insulate us if there's any issues with enabling it. 

Also, do we want to release v2 _and_ metrics at the same time? We could do one then the other? Again, may help us if there's any issues in the implementation

Reply:
> we would just set the variable in the AAS extension

I thought we wanted to enable by default, and release a new version if we see any issues. We could disable V2, metrics or both.

> Also, do we want to release v2 and metrics at the same time? 

I'm not against rolling out gradually but do we really need to? What would you have done exactly? You would have left disabled, release AAS with a flag to enable V2, waited for a week and released another AAS with metrics enabled?

Reply:
> I thought we wanted to enable by default, and release a new version if we see any issues. We could disable V2, metrics or both.

We could go either way

> What would you have done exactly? You would have left disabled, release AAS with a flag to enable V2, waited for a week and released another AAS with metrics enabled?

Exactly 🙂 

> Andrew wins!

You mean I'm to blame for missing OKRs 😂